crypto/evp/kdf_lib.c

```diff
@@ -144,6 +145,164 @@ int EVP_KDF_derive(EVP_KDF_CTX *ctx, unsigned char *key, size_t keylen,
     return ctx->meth->derive(ctx->algctx, key, keylen, params);
 }
 
+static int convert_key_cb(const OSSL_PARAM params[], void *arg)
+{
+    OSSL_PARAM_BLD *tmpl = arg;
+    const OSSL_PARAM *raw_bytes;
+    unsigned char *key_data;
+    size_t key_length;
+
+    raw_bytes = OSSL_PARAM_locate_const(params, OSSL_SKEY_PARAM_RAW_BYTES);
+    if (raw_bytes == NULL)
+        return 0;
+
+    if (!OSSL_PARAM_get_octet_string(raw_bytes, (void **)&key_data, 0,
+                                     &key_length)) {
+        return 0;
+    }
+
+    return ossl_param_build_set_octet_string(tmpl, NULL, OSSL_KDF_PARAM_KEY,
+                                             key_data, key_length);
+}
+
+int EVP_KDF_CTX_set_skey(EVP_KDF_CTX *ctx, EVP_SKEY *key)
+{
+    if (ctx == NULL)
+        return 0;
+
+    if (ctx->meth->set_key != NULL) {
+        if (ctx->meth->prov != key->skeymgmt->prov) {
+            /* TODO: export/import dance */
```

Comment:
@simo5 can we borrow the export-import code from the cipher implementation?

Reply:
What about a helper function ?

Reply:
Yes, that what I mean